src/scala/microsoft-spark-2.3.x/src/main/scala/org/apache/spark/api/dotnet/DotnetBackendHandler.scala

```diff
@@ -65,24 +66,37 @@ class DotnetBackendHandler(server: DotnetBackend)
               logError(s"Removing $objId failed", e)
               writeInt(dos, -1)
           }
+        case "rmThread" =>
+          try {
+            assert(readObjectType(dis) == 'i')
+            val threadToDelete = readInt(dis)
+            ThreadPool.deleteThread(threadToDelete)
+            writeInt(dos, 0)
+            writeObject(dos, null)
+          } catch {
+            case e: Exception =>
+              logError(s"Removing thread $threadId failed", e)
+              writeInt(dos, -1)
+          }
         case "connectCallback" =>
           assert(readObjectType(dis) == 'c')
           val address = readString(dis)
           assert(readObjectType(dis) == 'i')
           val port = readInt(dis)
-          DotnetBackend.setCallbackClient(address, port);
+          ThreadPool.run(threadId, () => DotnetBackend.setCallbackClient(address, port))
```

Comment:
We should be okay not running this part and "closeCallback" in the threadpool.

Reply:
Removed `ThreadPool` call.